timesketch/lib/analyzers/crash.py

```diff
@@ -0,0 +1,178 @@
+        """
+        self.index_name = index_name
+        super(CrashesSketchPlugin, self).__init__(index_name, sketch_id)
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query_elements = {
+            'Event - App Error' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"Application Error"',
+                'event_identifier:"1000"',
+                'event_level:"2"', # Level: Error
+            ),
+            'Event - WER' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"Windows Error Reporting"',
+                'event_identifier:"1001"',
+                'event_level:"4"', # Level: Info
+            ),
+            'Event - BSOD' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"Microsoft-Windows-WER-SystemErrorReporting"',
+                'event_identifier:"1001"',
+                'event_level:"2"', # Level: Error
+            ),
+            'Event - App Hang' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"Application Error"',
+                'event_identifier:"1002"',
+                'event_level:"2"', # Level: Error
+            ),
+            'Event - EMET Warning' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"EMET"',
+                'event_identifier:"1"',
+                'event_level:"3"', # Level: Warning
+            ),
+            'Event - EMET Error' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:"EMET"',
+                'event_identifier:"1"',
+                'event_level:"2"', # Level: Error
+            ),
+            'Event - .NET App Crash' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:".NET Runtime"',
+                'event_identifier:"1026"',
+                'event_level:"2"', # Level: Error
+            ),
+            'File - WER Report' : (
+                'data_type:"fs:stat"',
+                'filename:"/Microsoft/Windows/WER/"',
+                'filename:/((Non)?Critical|AppCrash)_.*/',
+                'file_entry_type:("directory" or "2")',
+            ),
+            'Registry - Crash Reporting' : (
+                'data_type:"windows:registry:key_value"',
+                r'key_path:r"\\Control\\CrashControl"',
+                'values:("LogEvent: REG_DWORD_LE 0" OR "SendAlert: REG_DWORD_LE 0" OR "CrashDumpEnabled: REG_DWORD_LE 0")',
+            ),
+            'Registry - Error Reporting' : (
+                'data_type:"windows:registry:key_value"',
+                r'key_path:"\\Software\\Microsoft\\PCHealth\\ErrorReporting"',
+                'values:("DoReport: REG_DWORD_LE 0" OR "ShowUI: REG_DWORD_LE 0")',
+            ),
+        }
+
+        # Creator of nested Elastic Search queries.
```

Comment:
why define all of these functions inside the run function, and not just as a separate function?

also this comment is not needed, and all functions lack docstrings